action.yml

```diff
@@ -115,6 +119,39 @@ runs:
           echo "Skipping GoSec"
         fi
 
+    - name: "Run StaticCheck"
+      shell: bash
+      run: |
+        set -eux
+
+        if [ "${{ inputs.run-staticcheck }}" == "true" ]; then
+          cd /tmp
+          curl -fsSL https://api.github.com/repos/dominikh/go-tools/releases/latest \
+            | grep -E "browser_download_url.*staticcheck_linux_amd64.tar.gz\"$" \
+            | cut -d '"' -f 4 \
+            | xargs -L 1 curl -fsSL -o /tmp/staticcheck_linux_amd64.tar.gz
+          tar -xvf /tmp/staticcheck_linux_amd64.tar.gz staticcheck/staticcheck
+          chmod +x ./staticcheck/staticcheck
+          curl -fsSL https://api.github.com/repos/codacy/codacy-staticcheck/releases/latest \
+            | grep "browser_download_url" | grep -v "browser_download_url.*jar" \
+            | cut -d '"' -f 4 \
+            | xargs -L 1 curl -fsSL -o /tmp/bin/codacy-staticcheck
+          chmod +x ./bin/codacy-staticcheck
+          cd -
+
+          find . -type f -name go.mod -exec bash -c 'cd $(dirname $1); PKGS=$(go list ./...); /tmp/staticcheck/staticcheck -f json $PKGS' _ {} \; > staticcheck-out.json
+          /tmp/bin/codacy-staticcheck < staticcheck-out.json > codacy-out.json
```

Comment:
Will the file `codacy-out.json` colide with other files if I run multiple standalone tools?

Reply:
yup, but its ok, we use it immediatly